csp/templatetags/csp.py

```diff
@@ -19,10 +19,9 @@ def __init__(self, nodelist):
 
     def render(self, context):
         output = self.nodelist.render(context)
-        nonce = context.get('request')
+        request = context.get('request')
+        nonce = request.csp_nonce() if hasattr(request, 'csp_nonce') else ''
```

Comment:
`process_request` only provides an interface to generate a nonce, it is actually generated when something has called `csp_nonce()`. The additional `hasattr` check in that ternary assures that the template tag doesn't throw an error if you load the tag but do not install the middleware.